openverse_catalog/dags/maintenance/add_license_url.py

```diff
@@ -76,60 +78,85 @@ def update_license_url(postgres_conn_id: str, task: AbstractOperator) -> dict[st
         postgres_conn_id=postgres_conn_id,
         default_statement_timeout=PostgresHook.get_execution_timeout(task),
     )
-    license_map = get_reverse_license_path_map()
 
-    total_count = 0
-    total_counts = {}
-    for license_items, path in license_map.items():
-        license_name, license_version = license_items
-        logger.info(f"Processing {license_name} {license_version}, {license_items}.")
-        license_url = f"{base_url}{path}/"
-
-        select_query = dedent(
-            f"""
-            SELECT identifier FROM image
-            WHERE (
-            meta_data is NULL AND license = '{license_name}'
-            AND license_version = '{license_version}')
-            """
-        )
-        result = postgres.get_records(select_query)
-
-        if not result:
-            logger.info(f"No records to update with {license_url}.")
+    select_query = dedent(
+        """
+    SELECT identifier, license, license_version
+    FROM image WHERE meta_data IS NULL;"""
+    )
+    records_with_null_in_metadata = postgres.get_records(select_query)
+
+    # Dictionary with license pair as key and list of identifiers as value
+    records_to_update = {}
+
+    for result in records_with_null_in_metadata:
+        identifier, license_, version = result
+        license_pair = f"{license_},{version}"
+        if license_pair not in records_to_update:
+            records_to_update[license_pair] = [identifier]
+        else:
+            records_to_update[license_pair].append(identifier)
```

Comment:
This could be replaced with a [`defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict)!

```suggestion
    records_to_update = defaultdict(list)

    for result in records_with_null_in_metadata:
        identifier, license_, version = result
        license_pair = f"{license_},{version}"
        records_to_update[license_pair].append(identifier)
```

Reply:
Nice, thank you!